relay-wstring/src/lib.rs

```diff
@@ -370,6 +388,16 @@ impl fmt::Display for WStr {
     }
 }
 
+/// Whether a code unit is a single unit unicode code point.
+///
+/// UTF-16 can encode characters in either single units or pairs, where the first unit is a leading
+/// surrogate and the second unit is a trailing surrogate. Additionally, the entire range of
+/// `0xD800-0xDBFF` is reserved for surrogates. This means that single unit unicode code points are
+/// always encoded outside of this range.
+fn is_standalone_unit(code_unit: u16) -> bool {
+    code_unit < 0xD800 || code_unit > 0xDFFF
+}
```

Comment:
Having reviewed all the ranges I still don't see this.  I believe all the code units for which `is_standalone_unit(u) -> false` will have either `is_leading_surrogate(u) -> true` or `is_trailing_surrogate(u) -> true`.  So I'm not sure we needed to add this.

Reply:
Yes, you're right. For I've confused `0xD800` and `0xDB00` when scanning through the comments, hence some of the confusion. You're right that we don't need this.